### ISSUE TITLE
DateTieredCompaction for existing tsdb tables

### ISSUE THREAD

**Issue by churrodog** [OpenTSDB/opentsdb#1065]
Sometimes a user might not want to re-write all their existing tsdb data but would still like to utilize date tiered compaction queries for tsdb. 

Here is a quick proposal for having users with existing data in their tsdb tables utilize date tiered compaction without re-writing everything.

When we turn on the date tiered compaction configuration option from the client, we also have an optional timestamp parameter.  If the user provides the time stamp, we add that to the tsdb table's column family attributes.  

So now whenever the client makes a query, it will check to see if the min ts of that query is >= the timstamp set on the table.  If so, set the time range attribute on the scan, if not then ignore the time range attribute for the scan (like we do for default compactions).  If the optional timestamp is not set on the table attribute, then we can assume that ts = 0;

This would be really helpful for the folks at salesforce as their tsdb cluster is quite large and re-writing existing data is not possible.  But they want to utilize the date tiered compaction feature of tsdb.  This would allow folks to utilize this feature without re-writing their data.

**Comment by churrodog** [comment]
@karanmehta93 FYI

**Comment by manolama** [comment]
Hmm, couldn't we just put the timestamp in the config file for OpenTSDB? And I assume that if a query spans the time boundary then we'd just leave off the TS comparison and it would query normally?

**Comment by churrodog** [comment]
Agreed, that will be easier to implement.  I'll put up a pull request with the patch.  Thanks.

**Comment by churrodog** [comment]
https://github.com/OpenTSDB/opentsdb/pull/1071

**Comment by karanmehta93** [comment]
LGTM, except the file `test/core/TestTsdbTSConfig.java` has a lot of unnecessary diff generated.

**Comment by churrodog** [comment]
@manolama what do you think about this patch?  I'll fix the formatting issues but agree with your suggestion in going the configuration route.  

Thanks

**Comment by manolama** [comment]
Merging after the cherry-pick passes.

**Comment by IDerr** [comment]
If it is merged, is it possible to close this issue @manolama ?

Thanks for your work :D
